swiggy-pdp-cart/src/components/Main/Content/Content.js

```diff
@@ -0,0 +1,34 @@
+import React, { useEffect, useState } from "react";
+import Menu from "./Menu/Menu";
+import Cart from "./Cart/Cart";
+import Categories from "./Categories/Categories";
+import { fetchCategoriesList,fetchMenuList } from "../../../services/fakeFetchFunctions";
+import { categoryNameById } from "./helper/helper";
+import ContentCSS from './content.module.css'
+export default function Content(){
+  
+  const [category,setCategory]=useState('recommended');
+  const [menu,setMenu]=useState([]);
+  const [categoryList,setCategoryList]=useState([]);
+  
+  useEffect(() => {
+    const menu = fetchMenuList();
+    const categoryList = fetchCategoriesList();
+    setMenu(menu);
```

Comment:
You need to maintain a loading state for each of the API call separately and show loaders both in categories as well as menu.

For now, we can make do with one generic loader - so if either of categories or menu items is still loading, we can just show a Loading... text on the page.

Read about how to create custom hooks and try to come up with your own hooks so that you can encapsulate the implementation details and just return the data points required by the consumer

Example:
`const { categories, areCatetoriesLoading } = useCategories();`